File: 0001_0100/0074/0074.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, k = (int(x) for x in f.read().split())
    s = 0
    if k == 1:
        s = n
    else:
        while k % 2:
            n //= 2
            s += n
            k = k // 2 + 1
    q.write(str(s + k // 2))

File: 0001_0100/0074/0074.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, k, s int
	fmt.Fscan(f, &n, &k)
	if k == 1 {
		s = n
	} else {
		for k%2 == 1 {
			n /= 2
			s += n
			k = k/2 + 1
		}
	}
	fmt.Fprint(q, s+k/2)
}
